ivadomed/utils.py

```diff
@@ -619,6 +638,123 @@ def threshold_predictions(predictions, thr=0.5):
     return thresholded_preds
 
 
+def segment_volume(model_fname, model_metadata_fname, image_fname, roi_fname=None):
+    """Segment an image.
+
+    Segment an image (image_fname) using a already trained model (model_fname) given its
+    training parameters (model_metadata_fname). If provided, a RegionOfInterest (roi_fname)
+    is used to crop the image prior to segment it.
+
+    Args:
+        model_fname (string): model filename (.pt) to use.
+        model_metadata_fname (string): Configuration file filename (.json) used for the training,
+            see https://github.com/neuropoly/ivado-medical-imaging/wiki/configuration-file
+        image_fname (string): image filename (e.g. .nii.gz) to segment.
+        roi_fname (string): Binary image filename (e.g. .nii.gz) defining a region of interest,
+            e.g. spinal cord centerline, used to crop the image prior to segment it if provided.
+    Returns:
+        nibabelObject: Object containing the soft segmentation.
+
+    """
+    # Define device
+    device = torch.device("cpu")
+
+    # Load model training config
+    with open(model_metadata_fname, "r") as fhandle:
+        context = json.load(fhandle)
+
+    # If ROI is not provided then force center cropping
+    if roi_fname is None and 'ROICrop2D' in context["transformation_validation"].keys():
+        context["transformation_validation"] = dict((key, value) if key != 'ROICrop2D'
+                                                        else ('CenterCrop2D', value)
+                                                        for (key, value) in context["transformation_validation"].items())
+
+    # Transforms
```

Comment:
i've already seen that code somewhere ;-) 
https://github.com/neuropoly/ivado-medical-imaging/blob/master/ivadomed/main.py#L94

Reply:
Great point, new function implemented: `compose_transforms`.